File: 2023-12-16/5957.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt';
const [L, ...datas] = fs.readFileSync(filePath).toString().trim().split('\n');

const solution = (N, list) => {
  const dirtyStack = Array.from({ length: N }, (_, i) => N - i);
  const washStack = [];
  const dryStack = [];

  for (let i = 0; i < list.length; i++) {
    let [cmd, num] = list[i];
    for (let n = 0; n < num; n++) {
      if (cmd === 1) {
        washStack.push(dirtyStack.pop());
      } else {
        dryStack.push(washStack.pop());
      }
    }
  }
  console.log(dryStack.reverse().join('\n'));
};

solution(
  Number(L),
  datas.map((x) => x.split(' ').map(Number))
);

File: 2023-12-16/5957.py
N = int(input())

dirtyStack = [i for i in range(N, 0, -1)]
washStack, dryStack = [], []
while True:
    cleanCmd, cleanNum = map(int, input().split())
    for _ in range(cleanNum):
        if cleanCmd == 1:
            washStack.append(dirtyStack.pop())
        else:
            dryStack.append(washStack.pop())
    if len(dryStack) == N:
        break
    
print(*dryStack[::-1], sep='\n')
